From 8d135b080202ccd011a5b39365a777042e84c9f4 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:45 -0400
Subject: [fs] nfsd: move proc_compound xdr encode init to helper

Message-id: <1403560314-32525-104-git-send-email-bfields@redhat.com>
Patchwork-id: 84503
O-Subject: [Patch RHEL7 103/172] nfsd4: move proc_compound xdr encode init to helper
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Mechanical transformation with no change of behavior.

Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 2d124dfaad08733077d2c46755716b2873af65a3
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index dc0edc0..7d09498 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -1252,6 +1252,18 @@ static bool need_wrongsec_check(struct svc_rqst *rqstp)
  return !(nextd->op_flags & OP_HANDLES_WRONGSEC);
 }
 
+static void svcxdr_init_encode(struct svc_rqst *rqstp,
+          struct nfsd4_compoundres *resp)
+{
+ struct xdr_stream *xdr = &resp->xdr;
+ struct xdr_buf *buf = &rqstp->rq_res;
+ struct kvec *head = buf->head;
+
+ xdr->buf = buf;
+ xdr->p   = head->iov_base + head->iov_len;
+ xdr->end = head->iov_base + PAGE_SIZE;
+}
+
 /*
  * COMPOUND call.
  */
@@ -1269,13 +1281,10 @@ nfsd4_proc_compound(struct svc_rqst *rqstp,
  u32  plen = 0;
  __be32  status;
 
- resp->xdr.buf = &rqstp->rq_res;
- resp->xdr.p = rqstp->rq_res.head[0].iov_base +
-      rqstp->rq_res.head[0].iov_len;
+ svcxdr_init_encode(rqstp, resp);
  resp->tagp = resp->xdr.p;
  /* reserve space for: taglen, tag, and opcnt */
  resp->xdr.p += 2 + XDR_QUADLEN(args->taglen);
- resp->xdr.end = rqstp->rq_res.head[0].iov_base + PAGE_SIZE;
  resp->taglen = args->taglen;
  resp->tag = args->tag;
  resp->opcnt = 0;
-- 
1.7.1